src/scipp/curve_fit.py

```diff
@@ -134,18 +134,18 @@ def _prepare_numpy_outputs(da, params, map_over):
 
 def _make_defaults(f, coords, params):
     spec = getfullargspec(f)
-    all_args = {*spec.args, *spec.kwonlyargs}
-    if not set(coords).issubset(all_args):
-        raise ValueError("Function must take the provided coords as arguments")
-    default_arguments = dict(
-        zip(spec.args[-len(spec.defaults) :], spec.defaults, strict=True)
-        if spec.defaults
-        else {},
-        **(spec.kwonlydefaults or {}),
+    non_default_args = (
+        spec.args[: -len(spec.defaults)] if spec.defaults is not None else spec.args
+    )
+    args = (
+        {*non_default_args, *spec.kwonlyargs}
```

Comment:
```suggestion
    args = (
        {*spec.args, *spec.kwonlyargs}
```
given that the defaults are removed below.